docs/developer/debugging.md

```diff
@@ -0,0 +1,109 @@
+# Debugging
+
+## Debugging NGF Remotely in Kubernetes
+
+This section will walk you through how to attach an ephemeral [dlv](https://github.com/go-delve/delve) debugger
+container to NGF while it's running in Kubernetes. This will allow you to remotely debug NGF running in Kubernetes
+using your IDE.
+
+- Create a `kind` cluster:
+
+   ```makefile
+   make create-kind-cluster
+   ```
+
+- Build debug images and install NGF on your kind cluster:
+
+  - **For NGINX OSS:**
+
+    ```console
+    make GOARCH=$GOARCH debug-install-local-build
+    ```
+
+  - **For NGINX Plus:**
+
+    ```console
+    make GOARCH=$GOARCH debug-install-local-build-with-plus
+    ```
+
+  > Note: The default value of GOARCH in the [Makefile](/Makefile) is `amd64`. If you try and debug an amd64 container on an ARM machine you will see the following error in the dlv container logs: `could not attach to pid <pid>: function not implemented`.
+  > This is a known issue and the only workaround is to create an arm64 image by specifying `GOARCH=arm64` the above commands.
+  > For more information, see this [issue](https://github.com/docker/for-mac/issues/5191)
+
+- Start kubectl proxy in the background:
+
+  ```console
+  kubectl proxy &
+  ```
+
+- Save the NGF Pod name:
+
+  ```console
+  POD_NAME=<NGF Pod>
+  ```
+
+- Run the following curl command to create an ephemeral debug container:
```

Comment:
just to double check, kubectl debug will not work?

Reply:
Correct. With Kubectl debug, there's no way to set the security context of the ephemeral pod.